File: AOJ/0394/pilling_blocks.py
n,k = map(int, input().split())
res = 0
row = 1
w = 0
while n >= row :
    if row * k >= w :
        res += 1
        w += row
        n -= row
    elif n >= row + 1 : row += 1
    else : break
print(res)

File: AOJ/0394/pilling_blocks.java
import java.util.Scanner;

class Main {
  void solve(){
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt(), k = sc.nextInt();

    int res = 0, row = 1, w = 0;
    while ( n >= row ) {
      if ( row * k >= w ) {
        ++res;
        w += row;
        n -= row;
      }
      else if ( n >= row + 1 ) ++row;
      else break;
    }
    
    System.out.println(res);
  }

  public static void main(String[] args) { new Main().solve(); }
}
